aip/0143.md

```diff
@@ -0,0 +1,93 @@
+---
+aip:
+  id: 143
+  state: reviewing
+  created: 2019-07-24
+permalink: /143
+redirect_from:
+  - /0143
+---
+
+# Standardized codes
+
+Many common concepts, such as spoken languages, countries, currency, and so on,
+have common codes (usually formalized by the [International Organization for
+Standardization][iso]) that are used in data communication and processing.
+These codes address the issue that there are often different ways to express
+the same concept in written language (for example, "United States" and "USA",
+or "Espanol" and "Spanish").
+
+## Guidance
+
+For concepts where a standardized code exists and is in common use, fields
+representing these concepts **should** use the standardized code for both input
+and output.
+
+```proto
+// A message representing a book.
+message Book {
+  // Other fields...
+
+  // The IETF BCP-47 language code representing the lanaguge in which
+  // the book was originally written.
+  // https://en.wikipedia.org/wiki/IETF_language_tag
+  string language_code = 99;
+}
+```
+
+- Fields representing standardized concepts **must** use the appropriate data
```

Comment:
Add that field comments **must** indicate the standard they're following? I know it sounds obvious, but...

Reply:
Done.